File: app/src/main/java/com/mnj/java_solutions/gfg/FrequencyGame.java
package com.mnj.java_solutions.gfg;

import java.util.HashMap;
import java.util.Map;

public class FrequencyGame {

    public void LargeButMinFreq() {

        int[] arr = {2, 2, 5, 50, 1};
//        2,2;
//        5,1;
//        50,1;
//        1,1l

        HashMap<Integer, Integer> map = new HashMap<>();

        int min_value = Integer.MAX_VALUE;
        int ans = Integer.MIN_VALUE;

        int i = 0;
        for (int ele : arr) {
            map.put(ele, map.getOrDefault(ele, 0) + 1);
        }

        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {

            int value = entry.getValue();
            int key = entry.getKey();

            if (value <= min_value) {
                ans = Math.max(ans, key);
                min_value = value;
            }

        }
        System.out.println("==>> LargeButMinFreq :"+ans);

    }
}

File: app/src/main/java/com/mnj/java_solutions/gfg/FrequencyGame.kt
package com.mnj.java_solutions.gfg

import java.util.HashMap

class FrequencyGame {
    fun LargeButMinFreq() {
        val arr = intArrayOf(2, 2, 5, 50, 1)
        //        2,2;
//        5,1;
//        50,1;
//        1,1l
        val map = HashMap<Int, Int>()
        var min_value = Int.MAX_VALUE
        var ans = Int.MIN_VALUE
        val i = 0
        for (ele in arr) {
            map[ele] = map.getOrDefault(ele, 0) + 1
        }
        for ((key, value) in map) {
            if (value <= min_value) {
                ans = Math.max(ans, key)
                min_value = value
            }
        }
        println("==>> LargeButMinFreq :$ans")
    }
}